Assets/Scripts/PlayerCharacter/PlayerGun.cs

```diff
@@ -0,0 +1,116 @@
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace Cosmobot
+{
+    public class PlayerGun : MonoBehaviour, DefaultInputActions.IPlayerGunActions
+    {
+
+        [SerializeField] private PlayerCamera playerCamera;
+        [SerializeField] private Transform cameraTransform;
+        [SerializeField] private float maxPickupRange; // relative to character, not camera
+        [SerializeField] private float carryHoldingDistance;
+        [SerializeField] private float holdingDistanceZoomMultiplier = 1f;
+        [SerializeField] private Vector3 carryPositionFirstPersonOffset;
+        [SerializeField] private Vector3 carryPositionThirdPersonOffset;
+        [SerializeField] private float carryForceMultiplier = 1f;
+        [SerializeField] private string playerLayerName = "Player";
+        [SerializeField] private string itemLayerName = "Item";
+
+        private bool carryingItem = false;
+        private Transform carriedItemTransform = null;
+        private Rigidbody carriedItemBody = null;
+
+        private Vector3 carryPosition;
+
+        private DefaultInputActions actions;
+
+        public void Update()
+        {
+            UpdateModel();
+        }
+
+        public void FixedUpdate()
+        {
+            var carryPosOffset = playerCamera.isFirstPerson ? carryPositionFirstPersonOffset : carryPositionThirdPersonOffset;
+            carryPosOffset = Quaternion.Euler(0, transform.rotation.eulerAngles.y, 0) * carryPosOffset;
+
+            var holdDistance = carryHoldingDistance * (playerCamera.isZoomed ? holdingDistanceZoomMultiplier : 1);
+
+            carryPosition = transform.position + cameraTransform.forward * holdDistance + carryPosOffset;
+            if (carryingItem)
+            {
+                var force = (carryPosition - carriedItemTransform.position) * carryForceMultiplier
+                    - carriedItemBody.velocity;
+                carriedItemBody.AddForce(force, ForceMode.VelocityChange);
+            }
+        }
+
+        private void UpdateModel()
+        {
+            transform.rotation = playerCamera.transform.rotation;
+        }
+
+        public void OnShoot(InputAction.CallbackContext context)
+        {
+            
+        }
+
+        private void SetCarriedItem(Transform newItemTransform)
+        {
+            if(newItemTransform != null)
```

Comment:
I would prefer `if (newItemTransform is not null)` as it is more explicit.
Add space after if

Reply:
in rare situations, `!= null` can behave in unexpected ways. This is because C# allows you to overload `!=` operator, and if you make some shitty code (which is common in Unity and unity assets) you can got `true` from `o != null` when `o is null`. Same applies to `==`.

So it is better to use `is null` or `is not null`